app/api/event_tag_endpoint.py

```diff
@@ -0,0 +1,40 @@
+from tracardi.service.storage.driver import storage
+from typing import List
+from fastapi import APIRouter, Depends, HTTPException
+from app.config import server
+from .auth.authentication import get_current_user
+from tracardi.domain.event_tag import EventTag
+
+router = APIRouter(
+    dependencies=[Depends(get_current_user)]
+)
+
+
+@router.post("/event/tag/add", tags=["event"], include_in_schema=server.expose_gui_api, response_model=dict)
+async def add_tags(tag_form: EventTag):
+    result = await storage.driver.tag.add(
+        event_type=tag_form.type,
+        tags=tag_form.tags
+    )
+    if result.errors:
+        raise HTTPException(status_code=500, detail=result.errors)
+    return {"OK": {"new": result.saved, "updated": 1-result.saved}}
+
+
+@router.delete("/event/tag/delete", tags=["event"], include_in_schema=server.expose_gui_api, response_model=dict)
+async def delete_tags(tag_form: EventTag):
+    total, removed, result = await storage.driver.tag.remove(
+        event_type=tag_form.type,
+        tags=tag_form.tags
+    )
+    if total == 0:
+        return {"OK": {"removed": removed, "total": total}} if result["result"] == "deleted" else \
```

Comment:
You got me wrong. In task description OK means the 200 OK status. So thre sould not be return 

```
{"OK": {"removed": removed, "total": total}} 
```

Only:

```
{"removed": removed, "total": total}
```

When everything is OK

And ERROR means 500 Internal ERROR

So there should not be ERROR key as well. You shoul THrow error.

Reply:
throw exception for total = 0 and else